src/languages/scala.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+  cascadingMatcher,
+} from '../util/nodeMatchers';
+import { NodeMatcherAlternative, ScopeType } from '../typings/Types';
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  // treating classes = classlike
+  class: ['class_definition', 'object_definition', 'trait_definition'],
+  className: ['class_definition[name]', 'object_definition[name]', 'trait_definition[name]'],
+
+  ifStatement: 'if_expression',
+
+  string: ['interpolated_string_expression', 'string'],
+  comment: 'comment',
+
+  // list.size(), does not count foo.size (field_expression), or foo size (postfix_expression)
+  functionCall: 'call_expression',
+  namedFunction: 'function_definition',
+  anonymousFunction: 'lambda_expression',
+
+  argumentOrParameter: argumentMatcher('arguments', 'parameters', 'class_parameters', 'bindings'),
+
+  name: ['*[name]', '*[pattern]'],
+  functionName: 'function_definition[name]',
+
+  type: leadingMatcher([
+    'upper_bound[type]',
+    'lower_bound[type]',
+    'view_bound[type]',
+    'context_bound[type]',
+    'val_definition[type]',
+    'val_declaration[type]',
+    'var_definition[type]',
+    'var_declaration[type]',
+    'type_definition',
+    'extends_clause[type]',
+    'class_parameter[type]',
+    'parameter[type]',
+    'function_definition[return_type]',
+    'typed_pattern[type]',
+    'binding[type]',
+
+  ], [':']),
+  value: leadingMatcher(['*[value]', '*[default_value]', 'type_definition[type]'], ['=']),
+  condition: conditionMatcher('*[condition]'),
+
+  // Scala features unsupported in Cursorless terminology
+  //  - Pattern matching
+
+  // Cursorless terminology not yet supported in this Scala implementation
+  /*
+    lists and maps basic definition are just function calls to constructors, eg List(1,2,3,4)
+    These types are also basically arbitrary, so we can't really hard-code them
```

Comment:
Why can't we hard-code them?  Not saying you need to do it in this PR, but this comment implies that we can't do it, whereas I thought we were just punting

Reply:
They are just classes, not a syntax / language construct like `[]` in JS or whatever. Scala comes with [a bunch of them](https://docs.scala-lang.org/overviews/collections-2.13/overview.html) but people will also use other random libraries as well.

Reply:
Interesting.  Almost feels like we need LSP support for this one?  Let's just call it out in the missing language features issue and discuss there